K8s-deployment/Charts/elk/elasticsearch/es-values.yml

```diff
@@ -1784,6 +1808,12 @@ metrics:
     ##   - myRegistryKeySecretName
     ##
     pullSecrets: []
+  ## @param metrics.annotations [object] Annotations for metrics
+  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+  ##
+  annotations:
+    helm.sh/hook: "post-install,post-upgrade"
```

Comment:
what is this doing? what is this hook for?

Reply:
These hooks signify that the metrics pod would be brought up only after the elasticsearch pods are created and upgraded only after elasticsearch pods are upgraded. [Refer](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks).

Reply:
okay. got it.